tendrl/node_agent/provisioner/ceph/plugins/ceph_installer.py

```diff
@@ -0,0 +1,153 @@
+
+import json
+import logging
+import urllib3
+
+from tendrl.node_agent.provisioner.ceph.provisioner_base import\
+    ProvisionerBasePlugin
+
+LOG = logging.getLogger(__name__)
+
+
+class CephInstallerPlugin(ProvisionerBasePlugin):
+
+    _MGET = 'GET'
+    _MPOST = 'POST'
+    _CEPH_INSTALLER_API_PORT = '8181'
+
+    def __init__(
+        self,
+    ):
+        self.http = urllib3.PoolManager()
+
+    def set_provisioner_node(self, provisioner_node):
+        self.provisioner_node = provisioner_node
+
+    def install_mon(self, mons):
+        url = 'http://localhost:%s/api/mon/install' % \
+            self._CEPH_INSTALLER_API_PORT
+        data = {
+            "calamari": False,
+            "hosts": mons,
+            "redhat_storage": False,
+            "redhat_use_cdn": True,
+            "verbose": False,
+        }
+        resp = self.http.request(
+            self._MPOST,
+            url,
+            body=json.dumps(data),
+            headers={'Content-Type': 'application/json'})
+        if resp.status == 200:
+            try:
+                res_data = json.loads(resp.data.decode('utf-8'))
+            except (TypeError, ValueError, UnicodeError) as e:
+                raise Exception(
+                    'Server response was not valid JSON: %r' % e)
+            return res_data['identifier']
```

Comment:
`identifier` here is job id for http request. We need to keep tracking the same and find if its completed.